nav2_simple_commander/nav2_simple_commander/robot_navigator.py

```diff
@@ -0,0 +1,408 @@
+        self.change_maps_srv = self.create_client(LoadMap, '/map_server/load_map')
+        self.clear_costmap_global_srv = self.create_client(
+            ClearEntireCostmap, '/global_costmap/clear_entirely_global_costmap')
+        self.clear_costmap_local_srv = self.create_client(
+            ClearEntireCostmap, '/local_costmap/clear_entirely_local_costmap')
+        self.get_costmap_global_srv = self.create_client(GetCostmap, '/global_costmap/get_costmap')
+        self.get_costmap_local_srv = self.create_client(GetCostmap, '/local_costmap/get_costmap')
+
+    def setInitialPose(self, initial_pose):
+        self.initial_pose_received = False
+        self.initial_pose = initial_pose
+        self._setInitialPose()
+
+    def goThroughPoses(self, poses):
+        # Sends a `NavThroughPoses` action request
+        self.debug("Waiting for 'NavigateThroughPoses' action server")
+        while not self.nav_through_poses_client.wait_for_server(timeout_sec=1.0):
+            self.info("'NavigateThroughPoses' action server not available, waiting...")
+
+        goal_msg = NavigateThroughPoses.Goal()
+        goal_msg.poses = poses
+
+        self.info('Navigating with ' + str(len(goal_msg.poses)) + ' goals.' + '...')
+        send_goal_future = self.nav_through_poses_client.send_goal_async(goal_msg,
+                                                                         self._feedbackCallback)
+        rclpy.spin_until_future_complete(self, send_goal_future)
+        self.goal_handle = send_goal_future.result()
+
+        if not self.goal_handle.accepted:
+            self.error('Goal with ' + str(len(poses)) + ' poses was rejected!')
+            return False
+
+        self.result_future = self.goal_handle.get_result_async()
+        return True
+
+    def goToPose(self, pose):
+        # Sends a `NavToPose` action request
+        self.debug("Waiting for 'NavigateToPose' action server")
+        while not self.nav_to_pose_client.wait_for_server(timeout_sec=1.0):
+            self.info("'NavigateToPose' action server not available, waiting...")
+
+        goal_msg = NavigateToPose.Goal()
+        goal_msg.pose = pose
+
+        self.info('Navigating to goal: ' + str(pose.pose.position.x) + ' ' +
+                      str(pose.pose.position.y) + '...')
+        send_goal_future = self.nav_to_pose_client.send_goal_async(goal_msg,
+                                                                   self._feedbackCallback)
+        rclpy.spin_until_future_complete(self, send_goal_future)
+        self.goal_handle = send_goal_future.result()
+
+        if not self.goal_handle.accepted:
+            self.error('Goal to ' + str(pose.pose.position.x) + ' ' +
+                           str(pose.pose.position.y) + ' was rejected!')
+            return False
+
+        self.result_future = self.goal_handle.get_result_async()
+        return True
+
+    def followWaypoints(self, poses):
+        # Sends a `FollowWaypoints` action request
```

Comment:
A docstring here instead of the comment would enable any python IDE to pick this up in the autocompletes etc, 
```
"""
Sends a `FollowWaypoints` action request
"""
```
But again earlier style might dictate a comment here being the preferred choice.